java/core/src/java/org/apache/orc/impl/DynamicByteArray.java

```diff
@@ -294,6 +295,16 @@ public void setByteBuffer(ByteBuffer result, int offset, int length) {
     return result;
   }
 
+  public ByteBuffer get(int offset, int length) {
+    final int currentChunk = offset / chunkSize;
+    final int currentOffset = offset % chunkSize;
+    final int currentLength = Math.min(length, chunkSize - currentOffset);
+    if (currentLength == length) {
+      return ByteBuffer.wrap(data[currentChunk], currentOffset, length);
```

Comment:
This is the "win" right here.  If the data requested is contiguous within a single data chunk, then a view to the data can be returned without needing to copy into a buffer.